src/actions/StepPerformer.ts

```diff
@@ -19,9 +19,19 @@ export class StepPerformer {
     ) {
     }
 
+    extendJSContext(newContext: any): void {
+        for (const key in newContext) {
+            if (key in this.context) {
+                console.log(`Notice: Context ${key} is overridden by the new context value`);
+                break;
+            }
+        }
+        this.context = {...this.context, ...newContext};
+    }
```

Comment:
```ts
extendJSContext(newContext: any): void {
  const overriddenKey = Object.keys(newContext).find(key => key in this.context);
  
  if (overriddenKey) {
    console.log(`Notice: Context ${overriddenKey} is overridden by the new context value`);
  }
  
  this.context = {
    ...this.context,
    ...newContext,
  };
}
```

Reply:
Thinking about this now, we're not merging categories on the first initialization if user passes two categories with the same name.
It makes sense to me if we'll do the same thing there. But IDK, kind of an edge case that doesn't bother me much.